data/types/base.go

```diff
@@ -249,6 +249,61 @@ func (b *Base) Normalize(normalizer data.Normalizer) {
 	if b.Tags != nil {
 		sort.Strings(*b.Tags)
 	}
+	if b.Time != nil && *b.Time != "" {
+		parsingTimeFormat := "2006-01-02T15:04:05.999-0700"
+		parsedTime, err := time.Parse(TimeFormat, *b.Time)
+		if err != nil {
+			parsedTime, err = time.Parse(parsingTimeFormat, *b.Time)
+		}
+		if err == nil {
+			utcTimeString := parsedTime.UTC().Format(TimeFormat)
+			_, offset := parsedTime.Zone()
+			// Time field is not well formatted in UTC
+			if utcTimeString != *b.Time {
+				b.Time = pointer.FromString(utcTimeString)
+				// Time field was not set to UTC timezone
+				// we update zone name / zone offset
+				if utcTimeString != parsedTime.Format(TimeFormat) {
+					b.TimeZoneOffset = pointer.FromInt(offset / 60)
+				}
+			}
+			if b.TimeZoneOffset == nil {
+				offsetSet := false
+				if b.TimeZoneName != nil && *b.TimeZoneName != "" {
+					zoneLoc, err := time.LoadLocation(*b.TimeZoneName)
+					if err == nil {
+						_, offset := parsedTime.UTC().In(zoneLoc).Zone()
+						b.TimeZoneOffset = pointer.FromInt(offset / 60)
+						offsetSet = true
+					}
+				}
+				if offsetSet == false {
```

Comment:
```suggestion
				if b.TimeZoneOffset == nil {
```
The offsetSet variable is useless.